content/actions/use-cases-and-examples/building-and-testing/building-and-testing-rust.md

```diff
@@ -0,0 +1,146 @@
+
+## Introduction
+
+This guide shows you how to build, test, and publish a Rust package.
+
+{% data variables.product.prodname_dotcom %}-hosted runners have a tools cache with preinstalled software, which includes the dependencies for Rust. For a full list of up-to-date software and the preinstalled versions of Rust, see [AUTOTITLE](/actions/using-github-hosted-runners/about-github-hosted-runners#preinstalled-software).
+
+## Prerequisites
+
+You should already be familiar with YAML syntax and how it's used with {% data variables.product.prodname_actions %}. For more information, see [AUTOTITLE](/actions/using-workflows/workflow-syntax-for-github-actions).
+
+We recommend that you have a basic understanding of the Rust language. For more information, see [Getting started with Rust](https://www.rust-lang.org/learn).
+
+## Using a Rust workflow template
+
+{% data reusables.actions.workflow-templates-get-started %}
+
+{% data variables.product.prodname_dotcom %} provides a Rust workflow template that should work for most basic Rust projects. The subsequent sections of this guide give examples of how you can customize this workflow template.
+
+{% data reusables.repositories.navigate-to-repo %}
+{% data reusables.repositories.actions-tab %}
+{% data reusables.actions.new-starter-workflow %}
+1. The "Choose a workflow" page shows a selection of recommended workflow templates. Search for "Rust".
+1. Filter the selection of workflows by clicking **Continuous integration**.
+1. On the "Rust - by {% data variables.product.prodname_actions %}" workflow, click **Configure**.
+
+   ![Screenshot of the "Choose a workflow" page. The "Configure" button on the "Rust" workflow is highlighted with an orange outline.](/assets/images/help/actions/starter-workflow-rust.png)
+1. Edit the workflow as required. For example, change the version of Rust.
+1. Click **Commit changes**.
+
+{% ifversion fpt or ghec %}
+   The `rust.yml` workflow file is added to the `.github/workflows` directory of your repository.
+{% endif %}
+
+## Specifying a Rust version
+
+At the time of writing, the default rust compiler version is 1.83.0 rustup is available and can be used to install additional toolchains. For example, the following workflow temporarily sets the toolchain to nightly:
+
+```yaml copy
+      - name: Temporarily modify the rust toolchain version
+        run: rustup override set nightly
+      - name: Ouput rust version for educational purposes
+        run: rustup --version
+```
+
+### Caching dependencies
+
+You can cache and restore dependencies using the following example below. Note that you will need to have Cargo.lock in your repository to cache dependencies.
+
+```yaml copy
+      - name: ⚡ Cache
+        uses: actions/cache@v4
+        with:
+          path: |
+            ~/.cargo/registry
+            ~/.cargo/git
+            target
+          key: ${{ runner.os }}-cargo-${{ hashFiles('**/Cargo.lock') }}
```

Comment:
The `{{ }}` in this line seem to be causing the linter failure. We can avoid this by disabling Liquid processing for the line. See `{% raw %} in [Code blocks](https://docs.github.com/en/contributing/style-guide-and-content-model/style-guide#code-blocks).

```suggestion
          key: {% raw %}${{ runner.os }}-cargo-${{ hashFiles('**/Cargo.lock') }}{% endraw %}
```

The [error rending the new page](https://github.com/github/docs/actions/runs/12573366781/job/35046299814?pr=35725#step:5:21) might well be why it's not visible in the preview.

Reply:
Thanks a lot! that "fixed it", there seem to be other [problems](https://docs-35725-22a95e.preview.ghdocs.com/en/enterprise-cloud@latest/actions/use-cases-and-examples/building-and-testing/building-and-testing-rust): I'll look into that later